src/libmongoc/CMakeLists.txt

```diff
@@ -147,99 +135,180 @@ if (NOT ENABLE_ZSTD STREQUAL OFF)
-      # Sets OPENSSL_FOUND on success.
-      include (FindOpenSSL)
-   endif ()
+add_library(_mongoc-dependencies INTERFACE)
+add_library(mongo::detail::c_dependencies ALIAS _mongoc-dependencies)
+target_link_libraries(_mongoc-dependencies INTERFACE mongo::detail::c_platform)
+install(TARGETS _mongoc-dependencies EXPORT mongoc-targets)
+set_property(TARGET _mongoc-dependencies PROPERTY EXPORT_NAME detail::c_dependencies)
 
-   if (ENABLE_SSL STREQUAL LIBRESSL)
-      include (FindPkgConfig)
-      message (STATUS " Searching for LibreSSL/libtls")
-      pkg_check_modules (LIBRESSL libtls)
-      if (LIBRESSL_FOUND)
-         message (STATUS "   Found ${LIBRESSL_LIBRARIES}")
-         set (SSL_LIBRARIES ${LIBRESSL_LIBRARIES})
-         if (LIBRESSL_INCLUDE_DIRS)
-           include_directories (SYSTEM "${LIBRESSL_INCLUDE_DIRS}")
-         endif ()
-         link_directories ("${LIBRESSL_LIBRARY_DIRS}")
-         set (LIBRESSL 1)
-      else ()
-         message (STATUS "   Not found")
-      endif ()
-   endif ()
+# d888888b db      .d8888.
+# `~~88~~' 88      88'  YP
+#    88    88      `8bo.
+#    88    88        `Y8b.
+#    88    88booo. db   8D
+#    YP    Y88888P `8888Y'
+
+# Pick the TLS library to use.
+function(_pick_which_tls_lib ssl_opt)
+   # Initial value, if not later set:
+   set_property(DIRECTORY PROPERTY _mongoc-tls-backend OFF)
+   if(ssl_opt STREQUAL "AUTO")
+      # Try to pick one automatically
+      _auto_pick_tls_lib()
+   elseif(ssl_opt)
+      # The user named a TLS library specifically. Use that one:
+      _use_named_tls_lib("${ssl_opt}")
+   else()
+      # User has disabled TLS selection
+   endif()
+endfunction()
 
-   if (ENABLE_SSL STREQUAL DARWIN OR (APPLE AND ENABLE_SSL STREQUAL "AUTO"))
-      if (APPLE)
-         set (SECURE_TRANSPORT 1)
-      else ()
-         message (FATAL_ERROR "ENABLE_SSL=DARWIN only supported on Mac OS X")
-      endif ()
-   endif ()
+# Attempt to automatically find and use a TLS library for the current platform
+function(_auto_pick_tls_lib)
+   if(WIN32)
+      # Windows has SecureChannel
+      _use_named_tls_lib(WINDOWS)
+   elseif(APPLE)
+      # Apple has SecureTransport
+      _use_named_tls_lib(DARWIN)
+   else()
+      # Try to find one of the common open-source TLS libraries
+      find_package(OpenSSL)
+      if(OpenSSL_FOUND)
+         _use_named_tls_lib(OPENSSL)
+      else()
+         # No OpenSSL. What about LibreSSL?
+         find_package(LibreSSL)
+         if(LibreSSL_FOUND)
+            _use_named_tls_lib(LIBRESSL)
+         endif()
+      endif()
+   endif()
+endfunction()
 
-   if (ENABLE_SSL STREQUAL WINDOWS OR (WIN32 AND ENABLE_SSL STREQUAL "AUTO"))
-      if (WIN32)
-         set (SECURE_CHANNEL 1)
-      else ()
-         message (FATAL_ERROR "ENABLE_SSL=WINDOWS only supported on Windows")
-      endif ()
-   endif ()
+# Per-backend link libs/options:
+set(SecureTransport/LINK_OPTIONS "SHELL:-framework CoreFoundation" "SHELL:-framework Security")
```

Comment:
Remove `SHELL:`?

On local install (with macOS), this produced a package config file with `SHELL:` in `Libs`:

```
# pkg-config .pc file generated by CMake 3.27.4 for libmongoc-1.11.1. DO NOT EDIT!
prefix=/Users/kevin.albertson/review/mongo-c-driver-1408/.install
exec_prefix=${prefix}
libdir=${exec_prefix}/lib

Name: libmongoc
Description: The MongoDB C Database Driver library
Version: 1.11.1
Requires: libbson-static-1.0

Libs: -L${libdir} -lmongoc-static-1.0 -lresolv -lsasl2 -framework Corefoundation -framework Security -L/opt/homebrew/Cellar/zstd/1.5.5/lib -lzstd -L/Users/kevin.albertson/code/c-bootstrap/install/libmongocrypt-1.8.0-alpha1/lib -lmongocrypt -fsanitize=address,leak SHELL:-framework CoreFoundation SHELL:-framework Security
Cflags: -fPIC -fsanitize=address,leak -DMONGOC_STATIC -DBSON_STATIC -I${prefix}/include/libmongoc-1.0
```

Attempting to use resulted in an error:
```
% export PKG_CONFIG_PATH=/Users/kevin.albertson/review/mongo-c-driver-1408/.install/lib/pkgconfig
% /opt/homebrew/opt/llvm/bin/clang -o test.c $(pkg-config --libs --cflags libmongoc-static-1.0)
clang-16: error: no such file or directory: 'SHELL:-framework'
clang-16: error: no such file or directory: 'CoreFoundation'
clang-16: error: no such file or directory: 'SHELL:-framework'
clang-16: error: no such file or directory: 'Security'
```

Reply:
Reverted